### PR TITLE
Add 'Lead Priority' in Add Leads

### PR DESCRIPTION
**Pull request by subinasr**
Depends on https://github.com/the-deep/server/pull/477

### CODE REVIEW

File: src/views/AddLead/index.js [hunk 1]
```diff
@@ -640,14 +662,25 @@ class AddLead extends React.PureComponent {
...
                     </div>
-                    <SelectInput
-                        faramElementName="confidentiality"
-                        label={confidentialityInputLabel}
-                        options={isProjectSelected ? confidentiality : undefined}
-                        keySelector={confidentialityKeySelector}
-                        labelSelector={confidentialityLabelSelector}
-                        disabled={pendingLeadOptions || pending || !isProjectSelected}
-                    />
+                    <div className={styles.inlineGroup}>
```
Comment (AdityaKhatri) [inline src/views/AddLead/index.js:678]: @subinasr , looks like the style has not been included

**Review by AdityaKhatri — APPROVED** [review]

File: src/views/AddLead/index.js [hunk 2]
```diff
@@ -178,6 +199,7 @@ class AddLead extends React.PureComponent {
                 source: [requiredCondition],
                 authors: [],
                 confidentiality: [requiredCondition],
+                priority: [requiredCondition],
```
Comment (tnagorra) [inline src/views/AddLead/index.js:202]: Priority is required here but not required in Deep Client

File: src/views/AddLead/requests.js [hunk 3]
```diff
@@ -149,6 +149,7 @@ const requestOptions = {
         method: methods.POST,
         body: ({ params }) => ({
             ...params.values,
+            authors: isDefined(params.values.authors) ? params.values.authors : [],
```
Comment (tnagorra) [inline src/views/AddLead/requests.js]: Why wasn't author's default values set to []? Does faram clear out empty arrays?

Comment (AdityaKhatri) [reply src/views/AddLead/requests.js]: easier to handle it at one point instead of multiple

Comment (tnagorra) [reply src/views/AddLead/requests.js]: Actually, it's easier if you do it during initialization. An empty array can never be changed to undefined using MultiSelectInput.
But, should look into faram behavior

File: src/views/AddLead/index.js (hunk 2 above)
Comment (tnagorra) [reply src/views/AddLead/index.js:202]: @AdityaKhatri is this required?

Comment (AdityaKhatri) [reply src/views/AddLead/index.js:202]: It is not in the backend, but lets enforce it.
I've made necessary changes in client as well.

File: src/views/AddLead/requests.js (hunk 3 above)
Comment (AdityaKhatri) [reply src/views/AddLead/requests.js]: While getting the values from redux, if the url is not present it returns an empty object, the best place would be to do it.

Comment (tnagorra) [reply src/views/AddLead/requests.js]: why not add { author: [] } in default value in redux selector?

**Review by tnagorra — APPROVED** [review]

**Pull request merged**